src/components/custom-package-edit/custom-package-edit.js

```diff
@@ -0,0 +1,149 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { reduxForm } from 'redux-form';
+import isEqual from 'lodash/isEqual';
+
+import {
+  Button,
+  Icon,
+} from '@folio/stripes-components';
+import { processErrors } from '../utilities';
+
+import DetailsView from '../details-view';
+import PackageNameField, { validate as validatePackageName } from '../package-name-field';
+import PackageCoverageFields, { validate as validateCoverageDates } from '../package-coverage-fields';
+import DetailsViewSection from '../details-view-section';
+import NavigationModal from '../navigation-modal';
+import Toaster from '../toaster';
+import styles from './custom-package-edit.css';
+
+class CustomPackageEdit extends Component {
+  static propTypes = {
+    model: PropTypes.object.isRequired,
+    initialValues: PropTypes.object.isRequired,
+    handleSubmit: PropTypes.func,
+    onSubmit: PropTypes.func.isRequired,
+    pristine: PropTypes.bool
+  };
+
+  static contextTypes = {
+    router: PropTypes.shape({
+      history: PropTypes.shape({
+        push: PropTypes.func.isRequired
+      }).isRequired
+    }).isRequired,
+    queryParams: PropTypes.object
+  };
+
+  componentWillReceiveProps(nextProps) {
+    let wasPending = this.props.model.update.isPending && !nextProps.model.update.isPending;
+    let needsUpdate = !isEqual(this.props.initialValues, nextProps.initialValues);
+
+    if (wasPending && needsUpdate) {
+      this.context.router.history.push(
+        `/eholdings/packages/${this.props.model.id}${this.context.router.route.location.search}`,
+        { eholdings: true }
+      );
+    }
+  }
+
+  handleCancel = () => {
+    this.context.router.history.push(
+      `/eholdings/packages/${this.props.model.id}${this.context.router.route.location.search}`,
+      { eholdings: true }
+    );
+  }
+
+  render() {
+    let {
+      model,
+      handleSubmit,
+      onSubmit,
+      pristine
+    } = this.props;
+
+    let {
+      queryParams,
+      router
+    } = this.context;
+
+    let actionMenuItems = [
+      {
+        label: 'Cancel editing',
+        to: `/eholdings/packages/${model.id}${router.route.location.search}`
+      }
+    ];
+
+    if (queryParams) {
+      actionMenuItems.push({
+        label: 'Full view',
+        to: `/eholdings/packages/${model.id}/edit`
+      });
+    }
+
+    return (
+      <div>
```

Comment:
It'd be cool to use a fragment here. Try `<> </>` first and if that doesn't work: `React.Fragment`

Reply:
Oops, merged then saw this. I'll give that a shot on some follow-up work. I definitely like the reduced DOM.